doc/architecture/decisions/0015-container-api.md

```diff
@@ -0,0 +1,103 @@
+# 15. Container API
+
+Date: 2018-09-17
+
+## Status
+
+Proposed
+
+## Context
+
+* We have several different roles for using a Holochain app (through a *container*):
+  * local, QML based UI components
+  * as a special case of the above: administration UI
+  * externally located (living in a browser) web UIs connected through some form of IPC
+  * *bridged* apps, i.e. Holochain app as the user of another Holochain app
+  * A Holo light-client connecting to the Holochain instance running on a HoloPort full-node
+* With HoloSqape and the Holochain-nodejs, we already have two container implementations
+
+A Holochain app cannot be used directly. Holochain is built as a library that exposes an interface for
+creating Holochain instances and calling their zome functions (see [core_api](/core_api/src/lib.rs)).
+This *core_api* requires its client to provide a context, which holds the representation of the agent
+(name and keys), a logger and persistence and will also include a networking proxy in the future.
+
+While it is possible to use this library directly in a monolithic app where the UI is tightly
+coupled to the app and everything linked to the Holochain library into the same executable, we regard this
+as a special case, since this does not allow for app composability (which we think is a crucial
+concept for the Holochain app ecosystem). 
+
+Instead, in the context of the Rust iteration of Holochain, we have been following the notion of
+providing a *Container* as a relevant concept (and implemention) for the deployment of Holochain apps:
+we provide a Container (i.e. [HoloSqape](https://github.com/holochain/holosqape)) for each supported platform (Linux, MacOS, Windows, iOS, Android)
+that gets installed on a host machine. Holochain apps get installed into the container.
+The Container:  
+ * manages installed hApps (Holochain Apps),
+ * manages agent identities (keys) ,
```

Comment:
... always includes the core Dpki app and through it provides key management / assists with persisting encrypted agent identities?

Reply:
Hm, I guess that is right, that would make a lot of sense..